src/main/java/training/supportbank/dataconverters/CsvConverter.java

```diff
@@ -0,0 +1,41 @@
+package training.supportbank.dataconverters;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import training.supportbank.models.Transaction;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+// Converts CSV transaction data to Transaction objects
+public class CsvConverter {
+    private static final Logger logger = LogManager.getLogger();
+
+    public static List<Transaction> extractTransactions(BufferedReader reader) throws IOException {
+        String transactionLine;
+        List<Transaction> transactions = new ArrayList<>();
+
+        while ((transactionLine = reader.readLine()) != null) {
+            try {
+                String[] transactionProperties = transactionLine.split(",");
+                String stringDate = transactionProperties[0].replace("\"", "");
+                LocalDate date = LocalDate.parse(stringDate, DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+                String from = transactionProperties[1];
+                String to = transactionProperties[2];
+                String narrative = transactionProperties[3];
+                BigDecimal amount = new BigDecimal(transactionProperties[4].replace("\"", ""));
+                Transaction transaction = new Transaction(date, from, to, narrative, amount);
+                transactions.add(transaction);
+            } catch (NumberFormatException | DateTimeParseException e) {
+                logger.error(e.getMessage() + "\nTransaction omitted from calculation: " + transactionLine + "\n");
```

Comment:
All the `Logger` methods have an overload which take an exception parameter, so you can call this:

```
logger.error("some message", e);
```

rather than this:

```
logger.error(e.getMessage() + "some message");
```

It's less work and it means you don't lose the stack trace from the error. 🙂